etc/scripts/freeze_and_update_reqs.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import print_function
+
+import argparse
+import fnmatch
+from commoncode.fileutils import resource_iter
+import os
+import subprocess
+import shutil
+import sys
+
+python_version = str(sys.version_info[0]) + str(sys.version_info[1])
+py_abi = "{0}cp{1}{0}".format("*", python_version)
+
+
+def generate_req_text(input_dir, output_file=None, package_name=None):
+    """
+    Generate a requirement file at `output_file`(by default requirements.txt) of all dependencies wheels and sdists present in the `input_dir` directory.
+    If a `package_name` is provided it will be updated to its latest version.
+    """
+    thirdparty = resource_iter(input_dir, with_dirs=False)
+    # FIXME this code is for py 3.6 and later we will update for all version
+    dependencies = [
+        files
+        for files in thirdparty
+        if fnmatch.fnmatchcase(files, "*py3*")
+        or fnmatch.fnmatchcase(files, py_abi)
+        or (
+            fnmatch.fnmatchcase(files, "*tar.gz*")
+            and not fnmatch.fnmatchcase(files, "*py2-ipaddress-3.4.1.tar.gz*")
+        )
+    ]
+    if not (os.path.isdir("required_deps")):
+        os.mkdir("required_deps")
+    for deps in dependencies:
+        shutil.copy(deps, "required_deps")
+    subprocess.run(
+        [
+            "pip-compile",
+            "--generate-hashes",
+            "--find-links",
+            "required_deps",
+            "--upgrade",
+            "--output-file",
+            output_file,
+            "--verbose",
+            "--upgrade-package",
+            "package_name",
+            "--pip-args",
+            "--no-index",
+        ]
+    )
+    shutil.rmtree("required_deps")
+
+
+def main_with_args(args: str) -> None:
+    parser = argparse.ArgumentParser(
+        description="""Creates a archive for specific OS and specific python.
+EXAMPLE:
+freeze_and_update_reqs.py \\
+  --deps_directory DEPS_DIRECTORY \\
+  --output OUTPUT \\
+  --upgrade_package PACKAGE_NAME \\
+""",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--deps_directory",
+        help="Required: Thirdparty Dependencies directory to be archived.",
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        "--output",
+        help="Output file name. Required if more than one input file is given. Will be derived from input file otherwise.",
+        type=str,
+        default="requirements.txt",
+    )
+
+    parser.add_argument(
+        "--upgrade",
+        help="Upgrade all dependencies to new version.",
+        action="store_true",
+    )
+
+    parser.add_argument(
+        "--upgrade_package",
+        help="Specify particular packages to upgrade.",
+        type=str,
+        default=None,
+    )
+
+    args = parser.parse_args()
+
+    tpdir = args.deps_directory
+    output_file = args.output
+    package_name = args.upgrade_package
+    upgrade = args.upgrade or None
```

Comment:
This argument is not used at all... so what is it for?

Reply:
upgrade is for upgrading all packages whereas upgrade package is for specific package,

Reply:
But where do you use it?

Reply:
Is not an argument used anywhere in your functions

Reply:
It is an option just like ON/OFF . It is used here only , i have tested it it works, if you want i will paste output here.